tinyagentos/routes/peer.py

```
@@ -0,0 +1,342 @@
+
+    # Rate limit
+    if not _rate_limit_ok(contact_id):
+        raise HTTPException(status_code=429, detail="rate limit exceeded (60/min/contact)")
+
+    # Size limit — reject on Content-Length before we re-serialise.
+    cl = request.headers.get("content-length")
+    if cl is not None:
+        try:
+            cl_int = int(cl)
+        except ValueError:
+            raise HTTPException(status_code=400, detail="invalid Content-Length")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)